.github/workflows/images.yml

```
@@ -52,19 +55,27 @@ jobs:
           nix-shell --run "echo" shell.nix
 
       - name: Login to Docker Hub
+        if: ${{ github.event_name == 'push' }}
         uses: docker/login-action@v3
         with:
           username: ${{ secrets.DOCKERHUB_USERNAME }}
           password: ${{ secrets.DOCKERHUB_TOKEN }}
 
+      - name: Login to GHCR
+        if: ${{ github.event_name == 'workflow_dispatch' }}
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
       - name: Build and push the release images
         run: |
-          if [[ "${{ github.ref_type }}" == "tag" ]]; then
-            # Remove 'refs/tags/' prefix to get the tag name
-            TAG="${GITHUB_REF#refs/tags/}"
-            TAG="${TAG#v}"
+          if [[ "${{ github.event_name }}" == "workflow_dispatch" ]]; then
+            TAG="${{ inputs.tag }}"
             nix-shell --pure --run "./scripts/helm/update-version-upon-release.sh"
-          else
+            nix-shell --run "./scripts/release.sh --tag ${TAG} --registry ${{ inputs.namespace }}"
```

nit: you're already using inputs for the namespace
```suggestion
            nix-shell --run "./scripts/release.sh --tag ${{ inputs.tag }} --registry ${{ inputs.namespace }}"
```